File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/java/com/tencent/devops/scm/provider/git/tgit/auth/TGitUserPassAuthProviderAdapter.java
package com.tencent.devops.scm.provider.git.tgit.auth;

import com.tencent.devops.scm.api.pojo.auth.IScmAuth;
import com.tencent.devops.scm.api.pojo.auth.UserPassScmAuth;
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider;
import com.tencent.devops.scm.sdk.tgit.auth.TGitUserPassAuthProvider;

public class TGitUserPassAuthProviderAdapter implements TGitAuthProviderAdapter {

    @Override
    public boolean support(IScmAuth auth) {
        return auth instanceof UserPassScmAuth;
    }

    @Override
    public TGitAuthProvider get(IScmAuth auth) {
        UserPassScmAuth userPassGitAuth = (UserPassScmAuth) auth;
        return new TGitUserPassAuthProvider(userPassGitAuth.getUsername(), userPassGitAuth.getPassword());
    }
}

File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-tgit/src/main/kotlin/com/tencent/devops/scm/provider/git/tgit/auth/TGitUserPassAuthProviderAdapter.kt
package com.tencent.devops.scm.provider.git.tgit.auth

import com.tencent.devops.scm.api.pojo.auth.IScmAuth
import com.tencent.devops.scm.api.pojo.auth.UserPassScmAuth
import com.tencent.devops.scm.sdk.tgit.auth.TGitAuthProvider
import com.tencent.devops.scm.sdk.tgit.auth.TGitUserPassAuthProvider

/**
 * TGit 用户名密码授权提供者适配器
 */
class TGitUserPassAuthProviderAdapter : TGitAuthProviderAdapter {

    /**
     * 检查是否支持该授权类型
     * @param auth 授权信息
     * @return 当授权类型为UserPassScmAuth时返回true
     */
    override fun support(auth: IScmAuth): Boolean {
        return auth is UserPassScmAuth
    }

    /**
     * 获取用户名密码授权提供者
     * @param auth 授权信息
     * @return 用户名密码授权提供者实例
     */
    override fun get(auth: IScmAuth): TGitAuthProvider {
        val userPassGitAuth = auth as UserPassScmAuth
        return TGitUserPassAuthProvider(userPassGitAuth.username, userPassGitAuth.password)
    }
}
